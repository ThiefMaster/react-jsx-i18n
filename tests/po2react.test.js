import poToReact, {jsonifyMessage} from '../src/tools/po2react';


const expectJSON = (msg) => expect(jsonifyMessage(msg));


test('Messages are properly converted to JSON', () => {
    expectJSON('This is a test').toMatchSnapshot();
    expectJSON('This is a {link}test{/link}.').toMatchSnapshot();
    expectJSON('This is {dynamic}.').toMatchSnapshot();
    expectJSON('This is {dynamic} {link}link{/link}.').toMatchSnapshot();
    expectJSON('This is {dynamicString}.').toMatchSnapshot();
});


test('PO file is properly converted to JSON', () => {
    expect(poToReact('test-data/de_DE.po', 'messages-test')).toMatchSnapshot();
});
