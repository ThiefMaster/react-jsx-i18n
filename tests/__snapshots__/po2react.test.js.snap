// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages are properly converted to JSON 1`] = `"This is a test"`;

exports[`Messages are properly converted to JSON 2`] = `
Array [
  "Fragment",
  null,
  "This is a ",
  Array [
    "Param",
    Object {
      "name": "link",
    },
    "test",
  ],
  ".",
]
`;

exports[`Messages are properly converted to JSON 3`] = `
Array [
  "Fragment",
  null,
  "This is ",
  Array [
    "Param",
    Object {
      "name": "dynamic",
    },
  ],
  ".",
]
`;

exports[`Messages are properly converted to JSON 4`] = `
Array [
  "Fragment",
  null,
  "This is ",
  Array [
    "Param",
    Object {
      "name": "dynamic",
    },
  ],
  " ",
  Array [
    "Param",
    Object {
      "name": "link",
    },
    "link",
  ],
  ".",
]
`;

exports[`PO file is properly converted to JSON 1`] = `
Object {
  "messages-test": Object {
    "": Object {
      "domain": "messages-test",
      "lang": "de_DE",
      "plural_forms": "nplurals=2; plural=(n != 1)",
    },
    "\\"<strong>Rats.</strong>\\"": Array [
      "\\"<strong>Ratten.</strong>\\"",
    ],
    "<HTML> is unescaped when extracted": Array [
      "<HTML> wird unescaped extracted",
    ],
    "Bye World": Array [
      "Tschüss Welt",
    ],
    "Hello & World": Array [
      "Hallo & Welt",
    ],
    "Hey {name}, you want to {link}click me{/link} and you know it!": Array [
      Array [
        "Fragment",
        null,
        "Hey ",
        Array [
          "Param",
          Object {
            "name": "name",
          },
        ],
        ", du willst ",
        Array [
          "Param",
          Object {
            "name": "link",
          },
          "mich anklicken",
        ],
        " und weißt das auch!",
      ],
    ],
    "Hover me": Array [
      "Hover mich",
    ],
    "Little cats:": Array [
      "Junge Katzen:",
    ],
    "Little dogs:": Array [
      "Junge Hunde:",
    ],
    "Some <HTML> & entities: → &": Array [
      "Etwas <HTML> & Entities: → &",
    ],
    "This is an emphasized dynamic value: {number}": Array [
      Array [
        "Fragment",
        null,
        "Das ist ein hervorgehobener dynamischer Wert: ",
        Array [
          "Param",
          Object {
            "name": "number",
          },
        ],
      ],
    ],
    "This is an emphasized translated value: {tag}hello{/tag}": Array [
      Array [
        "Fragment",
        null,
        "Das ist ein hervorgehobener übersetzter Wert: ",
        Array [
          "Param",
          Object {
            "name": "tag",
          },
          "hallo",
        ],
      ],
    ],
    "You    are ugly!": Array [
      "Du    bist hässlich!",
    ],
    "You have {count} rat.": Array [
      Array [
        "Fragment",
        null,
        "Du hast ",
        Array [
          "Param",
          Object {
            "name": "count",
          },
        ],
        " Ratte.",
      ],
      Array [
        "Fragment",
        null,
        "Du hast ",
        Array [
          "Param",
          Object {
            "name": "count",
          },
        ],
        " Ratten.",
      ],
    ],
    "bigcat": Array [
      "KATZE",
      "KATZEN",
    ],
    "cat": Array [
      "Katze",
      "Katzen",
    ],
    "catoffspring": Array [
      "Kätzchen",
    ],
    "dogoffspring": Array [
      "Welpe",
    ],
    "foo   {weird} bar {hello}{test} xxx": Array [
      Array [
        "Fragment",
        null,
        "Foo   ",
        Array [
          "Param",
          Object {
            "name": "weird",
          },
        ],
        " Bar ",
        Array [
          "Param",
          Object {
            "name": "hello",
          },
        ],
        Array [
          "Param",
          Object {
            "name": "test",
          },
        ],
        " XXX",
      ],
    ],
    "foo: {foo}": Array [
      Array [
        "Fragment",
        null,
        "Foo: ",
        Array [
          "Param",
          Object {
            "name": "foo",
          },
        ],
      ],
    ],
    "foobar": Array [
      "FooBar",
    ],
    "params-testfoo: {foo}": Array [
      Array [
        "Fragment",
        null,
        "Foo: ",
        Array [
          "Param",
          Object {
            "name": "foo",
          },
        ],
      ],
    ],
    "xxx foo bar{test}{x} y{/x} moo": Array [
      Array [
        "Fragment",
        null,
        "xxx foo bar",
        Array [
          "Param",
          Object {
            "name": "test",
          },
        ],
        Array [
          "Param",
          Object {
            "name": "x",
          },
          " y",
        ],
        " muh",
      ],
    ],
  },
}
`;
